sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -176,17 +176,32 @@ typedef struct az_iot_hub_client_properties
 
 /**
  * @brief Initializes the Telemetry or C2D properties.
+ * 
+ * @note The properties init API will not encode properties. In order to support
+ *       the following characters: 
+ *          `/ % # &`
+ *       the onus is on the application to properly url encode them (RFC 2396). Only these characters

```

Comment:
Please rephrase to say "The application must encode the following characters." - What you wrote fails to translate in my native language via automatic translators.
Also, the RFC you quote is obsolete and url-encode is incorrect.
You could either:
1. Say "Percent-encode only the following characters as described in RFC 3986, section 2.2"
2. Replace the following characters with the following encoding (and for each of the 4 characters give the replacement formula).

I prefer 2 because the rules in the RFC require other characters to be percent encoded depending on the location within the URI.